### DISCUSSION TITLE
Facial recognition

### DISCUSSION THREAD

**Discussion by endaman** [immich-app/immich#6703] [Q&A] [answered]
on version 1.93,3
Is there a way to make facial recognition job faster?  Importing 80K images on home server, and has taken several days still not done yet.  Only 1 active job.  Can I increase facial recognition concurrency setting somehow?  It is currently greyed out and set to 1.
Thanks

**Comment by mertalev** [comment]
Is there only 1 active job and no waiting, or are there waiting jobs that gradually decrease?

**Comment by endaman** [comment]
1 active job and 17K waiting, slowly decreasing.  Just wondering if there's a setting that I missed to make this process faster.

**Reply by mertalev** [reply, marked as answer]
Nope. The speed depends a lot on your CPU and whether it has certain SIMD instructions. But it's also worth noting that the "waiting" number is a bit misleading because some jobs also requeue themselves so it'll look slower than it actually is. There will come a turning point where the "waiting" count starts dropping much more quickly.

**Reply by endaman** [reply]
> Nope. The speed depends a lot on your CPU and whether it has certain SIMD instructions. But it's also worth noting that the "waiting" number is a bit misleading because some jobs also requeue themselves so it'll look slower than it actually is. There will come a turning point where the "waiting" count starts dropping much more quickly.

Thank you.  Guess my old dual xeon E5-2690v4 server doesn't have it.
